rspace/src/main/scala/coop/rchain/rspace/history/KeySegment.scala

```diff
@@ -0,0 +1,47 @@
+package coop.rchain.rspace.history
+import scala.annotation.tailrec
+
+/**
+  * Segment of a History key
+  * @tparam V type of a key value
+  * @tparam I type of a word in key alphabet
+  */
+trait KeySegment[V, I] {
+  def size: Long
+  def isEmpty: Boolean
+  def nonEmpty: Boolean
+  def head: I
+  def headOption: Option[I]
+  def toHex: String
+  def :+(item: I): V
+  def tail: V
+  def ++(other: V): V
```

Comment:
What this trait represents? A type-class or the data itself.

If it's data itself tail should return `KeySegment` as the rest without head.

If it's a type-class each operation should accept data similar to static functions.

E.g. `toHex` looks like transforming KeySegment to a string so it's like KeySegment as data, but `++` accepts V so it's not combining two segments KeySegments.